library/ZendService/Twitter/Response.php

```diff
@@ -152,4 +160,52 @@ public function toValue()
     {
         return $this->jsonBody;
     }
+
+    /**
+     * Retun the array of response rate info
+     *
+     * @return array
+     */
+    public function getRateLimit()
+    {
+        return $this->rateLimit;
+    }
+
+    /**
+     * Populates the object with info. This can possibly called from the
+     * constructor, or it can be called later.
+     *
+     */
+    public function populate($httpResponse = null)
+    {
+
+        if (is_null($httpResponse)) {
+            $httpResponse = $this->httpResponse;
+        }
+
+        $this->httpResponse = $httpResponse;
+        $this->rawBody      = $httpResponse->getBody();
+
+        $headers = $this->httpResponse->getHeaders();
+
+        if (! is_null($headers)) {
+            $headers = $headers->toArray();
```

Comment:
The semantic meaning of `$headers` seems to change during the method execution, as you are getting some sort of parameter bag and later on possibly converting it to an array. Consider defining two variables instead: that would also allow you to drop the `else` statement